### PR TITLE
Backport "HBASE-28716 Users of QuotaRetriever should pass an existing connection (#6065)" to branch-3

### PR DESCRIPTION
**Pull request by ndimiduk**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ branch-3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  0s |  |  branch-3 passed  |
| +1 :green_heart: |  compile  |   3m 44s |  |  branch-3 passed  |
| +1 :green_heart: |  checkstyle  |   0m 57s |  |  branch-3 passed  |
| +1 :green_heart: |  spotbugs  |   2m 17s |  |  branch-3 passed  |
| +1 :green_heart: |  spotless  |   0m 46s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 43s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 43s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 55s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   2m 32s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 41s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 21s |  |  The patch does not generate ASF License warnings.  |
|  |   |  40m 43s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6096/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6096 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 27cd348f4052 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / 02d76d206d44556c44e3bfe2074e5dd1db1315ae |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 84 (vs. ulimit of 30000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6096/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 36s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ branch-3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  2s |  |  branch-3 passed  |
| +1 :green_heart: |  compile  |   1m 21s |  |  branch-3 passed  |
| +1 :green_heart: |  javadoc  |   0m 50s |  |  branch-3 passed  |
| +1 :green_heart: |  shadedjars  |   5m 22s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 57s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 21s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 21s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 48s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 19s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 36s |  |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 285m 56s |  |  hbase-server in the patch passed.  |
|  |   | 315m 16s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6096/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6096 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 4d05d0516350 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / 02d76d206d44556c44e3bfe2074e5dd1db1315ae |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6096/1/testReport/ |
| Max. process+thread count | 4870 (vs. ulimit of 30000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6096/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/quotas/QuotaRetriever.java [hunk 1]
```diff
@@ -55,19 +55,29 @@ public class QuotaRetriever implements Closeable, Iterable<QuotaSettings> {
   /**
    * Should QutoaRetriever manage the state of the connection, or leave it be.
    */
-  private boolean isManagedConnection = false;
+  private final boolean isManagedConnection;
 
-  QuotaRetriever() {
+  public QuotaRetriever(final Connection conn) throws IOException {
+    this(conn, (QuotaFilter) null);
```
Comment (pankaj72981) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/quotas/QuotaRetriever.java:61]: Ah, missed while reviewing the master branch PR. 

Nit: We dont need to type cast the NULL

**Review by pankaj72981 — APPROVED** [review]
LGTM

File: hbase-client/src/main/java/org/apache/hadoop/hbase/quotas/QuotaRetriever.java (hunk 1 above)
Comment (ndimiduk) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/quotas/QuotaRetriever.java:61]: I believe it is required. Without it, Java won't know which version of the two-arg constructor to call -- there are two variants that have `Connection` as the first argument. Indeed, IntelliJ complains when I remove this cast.

**Pull request merged**